scripts/main.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+type about struct {
+	Allowed     []string `yaml:"allowed"`
+	Comment     string   `yaml:"comment"`
+	DataType    string   `yaml:"datatype"`
+	Type        string   `yaml:"type"`
+	UUID        string   `yaml:"uuid"`
+	Description string   `yaml:"description"`
+}
+
+var dataTypes = map[string]string{
+	"boolean":  "bool",
+	"string":   "string",
+	"uint16":   "uint16",
+	"float":    "float64",
+	"int16":    "int16",
+	"double":   "float64",
+	"int32":    "int32",
+	"uint8":    "uint8",
+	"string[]": "[]string",
+	"uint8[]":  "[]uint8",
+	"int8":     "int8",
+	"uint32":   "uint32",
+}
+
+func main() {
+
+	arg := ""
+	if len(os.Args) > 1 {
+		arg = os.Args[1]
+	} else {
+		log.Fatal(errors.New("yaml file required"))
+	}
+
+	result := make(map[string]about)
+
+	yamlFile, err := ioutil.ReadFile(arg)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = yaml.Unmarshal(yamlFile, &result)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	orderedKeys := make([]string, len(result))
+
+	for k := range result {
+		orderedKeys = append(orderedKeys, k)
+	}
+
+	sort.Strings(orderedKeys)
+
+	output := ""
+	currentSubStruct := ""
+	prevLen := 0
+	parentArray := []string{"Vehicle"}
+
+	output = "package shared\n\ntype DataSchemaStruct struct {\n"
+
+	for _, k := range orderedKeys {
+
+		abt := result[k]
+		attrs := strings.Split(k, ".")
```

Comment:
I think I see why you're worried about sorting now. The [JSON output](https://github.com/COVESA/vehicle_signal_specification/releases/download/v3.0/vss_rel_3.0.json) is already hierarchical, maybe that's a better input?